src/lib/API/api.js

```diff
@@ -125,74 +125,88 @@ export class APIService {
     })())
   }
 
+  get = throttle((...params) => {
```

Comment:
1. add the following utility:

```javascript
// src/lib/util/axios.js
import { throttle } from 'lodash'

export const throttleCalls = (client, interval = 1000) => {
  ['get', 'post', 'put', 'delete'].forEach(method => {
    const originalMethod = client[method].bind(client)

    client[method] = throttle(originalMethod, interval)
  });

  return client
}

/* 
    as an option you could keep client  immutable just wrap it:

    import { throttle, fromPairs } from 'lodash'

    export const throttleCalls = (client, interval = 1000) => fromPairs(
      ['get', 'post', 'put', 'delete'].map(method => {
        const originalMethod = client[method].bind(client)

        return [method, throttle(originalMethod, interval)]
      })
    )
*/

```

2. import utility and replace line 123 with

```javascript
this.client = throttleCalls(instance)
```
also, `axios.create()` is a synchronous function so you don't need to `await` it

3. discard other changes in api.js

Reply:
do we really want to have a fixed interval for all requests in the system?
I think we might want a more granular approach, perhaps even set different interval per request

Reply:
@omerzam thats how it was implemented before.
i've reviewed the api calls we make and there's no reason not to  throttle for 1req per second.
also its throttled per http method get,post... each gets its  own throttle.
it just  simplify things.